### PR TITLE
Use activeCluster instead of currentCluster if there is only one cluster in namespace's cluster list

### PR DESCRIPTION
**Pull request by hai719**
## What changed?
Use activeCluster instead of currentCluster if there is only one cluster in namespace's cluster list

## Why?
- Workflow requests will get `workflow execution not found` error under the following scenario:
    - Namespace handover is completed and new workflows are created on new active cluster.
    - Old active cluster is removed from namespace. Namespace has only one cluster which is the new active.
    - Workflow requests hit old active cluster because DNS is not updated.

If there is only one entry in namespace’s cluster list, and if the entry is not local, we should forward the request to that cluster instead of handling locally.

## How did you test it?
unit test.

### CODE REVIEW

File: common/rpc/interceptor/dc_redirection_policy.go [hunk 1]
```diff
@@ -188,8 +188,7 @@ func (policy *SelectedAPIsForwardingRedirectionPolicy) getTargetClusterAndIsName
 	}
 
 	if len(namespaceEntry.ClusterNames()) == 1 {
-		// do not do dc redirection if namespace is only targeting at 1 dc (effectively local namespace)
-		return policy.currentClusterName, false
+		return namespaceEntry.ActiveClusterName(), true
```
Comment (yux0) [inline common/rpc/interceptor/dc_redirection_policy.go:191]: we can just remove this if?

Comment (hai719) [reply common/rpc/interceptor/dc_redirection_policy.go:191]: The following code can still return current non-active cluster if there is only 1 cluster in namespaceEntry.

Comment (yux0) [reply common/rpc/interceptor/dc_redirection_policy.go:191]: I think if the auto forwarding is not enabled, then we should still use current cluster?

Comment (hai719) [reply common/rpc/interceptor/dc_redirection_policy.go]: removed per offline discussion

**Pull request merged**